File: assets/code-variants/ch13/01.py
class PaymentProcessor:
    _credit_card_service: CreditCardService
    ...
    def make_payment(self, credit_card, amount):
        if credit_card.is_expired():
            return False
        success = self._credit_card_service.charge_credit_card(credit_card, amount)
        return success

File: assets/code-variants/ch13/01.java
class PaymentProcessor {
  private CreditCardService creditCardService;
  ...
  boolean makePayment(CreditCard creditCard, Money amount) {
    if (creditCard.isExpired()) { return false; }
    boolean success = creditCardService.chargeCreditCard(creditCard,  amount);
    return success;
  }
}
